src/Raven.Pal/src/posix/pager.c

```diff
@@ -48,9 +50,13 @@ int32_t rvn_lock_memory(int32_t open_flags, void *mem, int64_t size, int32_t *de
         goto Exit;
     }
 
-    if (mlock(mem, size) ||
+    if (!mlock(mem, size) ||
```

Comment:
I know this is C code, but we should decide if we keep the convention here too. Based on what we do on C# it should be `mlock(mem, size) != 0` instead.